### ISSUE TITLE
Add Opus support

### ISSUE THREAD

**Issue by berolinux** [icculus/SDL_sound#24]
SDL_sound currently supports Vorbis files -- but Xiph (maintainers of Vorbis) have deprecated Vorbis in favor of Opus back in 2013. It would be nice to have Opus support in SDL_sound as well.

**Comment by sezero** [comment]
As far as I know, @icculus is trying to avoid external library dependencies in SDL_sound-2.0: if you know of an opus decoder that is both mature and also suitably licensed for embedding in SDL_sound - such as those so-called one-header libs, please let us know.

**Comment by icculus** [comment]
https://nothings.org/stb/stb_opus.html

**Comment by feliwir** [comment]
A decoder implementation using libopus would not be accepted/viable?

**Comment by icculus** [comment]
> A decoder implementation using libopus would not be accepted/viable?

We are avoiding external dependencies beyond SDL2.

**Comment by sezero** [comment]
Close this one?

**Comment by feliwir** [comment]
Well, this is still a valid issue.
Maybe there will be a single file opus decoder at some day (not necessarly stb_opus), which can be used.

**Comment by icculus** [comment]
Yeah, leave it open for now, but it's not going to move any time soon.

**Comment by ericoporto** [comment]
Can't there be an exception in this case with the option of accepting an external dependency for libopus? The license at least looks alright: https://github.com/xiph/opus/blob/main/COPYING.
